File: 896-monotonic-array/896-monotonic-array.rb
# @param {Integer[]} nums
# @return {Boolean}
def is_monotonic(nums)
    store = 0
    ((nums.size)-1).times do |i|
        c = nums[i] - nums[i+1];
        c = (c < 0) ? -1 : (c > 0) ? 1 : 0;
        
        if c != 0
            if c != store and store != 0
                return false
            end
            store = c
        end
    end
    
    return true
end

File: 896-monotonic-array/896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        if (nums.size() == 1) return true;
        
        const int SIZE = nums.size();
        int i = 0;
        
        while(i < SIZE-1 && nums[i] == nums[i+1]) ++i;
        
        if(i == SIZE-1) return true;
        
        bool ans = nums[i] <= nums[i+1];
        for(i=i+1; i<SIZE-1; ++i) {
            if(nums[i] == nums[i+1]) continue;
            if(nums[i] < nums[i+1] != ans) return false;
            ans = nums[i] < nums[i+1];
        }
        
        return true;
    }
};